aea/helpers/dependency_tree.py

```diff
@@ -198,6 +198,8 @@ def generate(cls, packages_dir: Path) -> List[List[PackageId]]:
                     "name", item_config.get("agent_name")
                 )
                 public_id = PublicId.from_json(item_config)
+                if public_id.name == "scaffold":
+                    continue
```

Comment:
This is not the right place to ignore. The generate method's responsibility is neatly defined and should not have this custom exception.